runtime/src/qio/qbuffer.c

```diff
@@ -625,6 +625,7 @@ void qbuffer_reposition(qbuffer_t* buf, int64_t new_offset_start)
   while( ! deque_it_equals(iter, end) ) {
     qbp = (qbuffer_part_t*) deque_it_get_cur_ptr(sizeof(qbuffer_part_t), iter);
     qbp->end_offset += diff;
+    deque_it_forward_one(sizeof(qbuffer_part_t), &iter);
```

Comment:
I haven't looked at this code in a while, so I don't remember a lot of the details, but can you describe why you needed to add this? At a guess, I would think that this would advance the iterator to the next "part", but I'm not sure that is always appropriate here.

Reply:
Without that line, my test cases were getting stuck in the while loop forever. I figured the missing `forward_one` was a bug, and that the other two uses of this method in `qio_channel_seek` and `_qio_channel_makebuffer_unlocked` never ran into it because the buffer had just been cleared or initialized (meaning iter == end). 

If there are multiple parts, shouldn't this function always loop through all of them and shift their end offset?

Reply:
Oh, they don't have an `offset_start` anyway.. Carry on!